homeassistant/components/govee_light_local/__init__.py

```diff
@@ -14,14 +19,29 @@
 
 PLATFORMS: list[Platform] = [Platform.LIGHT]
 
+_LOGGER = logging.getLogger(__name__)
+
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Govee light local from a config entry."""
 
     coordinator: GoveeLocalApiCoordinator = GoveeLocalApiCoordinator(hass=hass)
-    entry.async_on_unload(coordinator.cleanup)
 
-    await coordinator.start()
+    async def await_cleanup():
+        cleanup_complete: asyncio.Event = coordinator.cleanup()
+        with suppress(TimeoutError):
+            await asyncio.wait_for(cleanup_complete.wait(), 1)
+
+    entry.async_on_unload(await_cleanup)
+
+    try:
+        await coordinator.start()
+    except OSError as ex:
+        if ex.errno != EADDRINUSE:
+            _LOGGER.error("Start failed, errno: %d", ex.errno)
+            return False
+        _LOGGER.error("Port %s already in use", LISTENING_PORT)
```

Comment:
Please don't both log and raise the exception. Pass the message as an argument to the exception instead. The message will be logged by default at debug level.